tools/clang/include/clang/SPIRV/Type.h

```diff
@@ -68,39 +71,51 @@ class Type {
   static const Type *getFloat16(SPIRVContext &ctx);
   static const Type *getFloat32(SPIRVContext &ctx);
   static const Type *getFloat64(SPIRVContext &ctx);
+
   static const Type *getVector(SPIRVContext &ctx, uint32_t component_type,
-                               uint32_t vec_size);
-  static const Type *getVec2(SPIRVContext &ctx, uint32_t component_type);
-  static const Type *getVec3(SPIRVContext &ctx, uint32_t component_type);
-  static const Type *getVec4(SPIRVContext &ctx, uint32_t component_type);
+                               uint32_t vec_size, DecorationSet decs = {});
```

Comment:
`Block` can only be applied to structure type.

Even things like `MatrixStride` (which as matrix in the name) cannot be applied to matrices. It is for structure type also.